checker/jdk/nullness/src/javax/script/ScriptEngineFactory.java

```diff
@@ -0,0 +1,231 @@
+     */
+    public List<String> getNames();
+
+    /**
+     * Returns the name of the scripting language supported by this
+     * <code>ScriptEngine</code>.
+     * @return The name of the supported language.
+     */
+    public String getLanguageName();
+
+    /**
+     * Returns the version of the scripting language supported by this
+     * <code>ScriptEngine</code>.
+     * @return The version of the supported language.
+     */
+    public String getLanguageVersion();
+
+    /**
+     * Returns the value of an attribute whose meaning may be implementation-specific.
+     * Keys for which the value is defined in all implementations are:
+     * <ul>
+     * <li>ScriptEngine.ENGINE</li>
+     * <li>ScriptEngine.ENGINE_VERSION</li>
+     * <li>ScriptEngine.LANGUAGE</li>
+     * <li>ScriptEngine.LANGUAGE_VERSION</li>
+     * <li>ScriptEngine.NAME</li>
+     * </ul>
+     * <p>
+     * The values for these keys are the Strings returned by <code>getEngineName</code>,
+     * <code>getEngineVersion</code>, <code>getLanguageName</code>,
+     * <code>getLanguageVersion</code> for the first four keys respectively. For NAME, one of the Strings
+     * returned by <code>getNames</code> is returned.<br><br>
+     * A reserved key, <code><b>THREADING</b></code>, whose value describes the behavior of the engine
+     * with respect to concurrent execution of scripts and maintenance of state is also defined.
+     * These values for the <code><b>THREADING</b></code> key are:<br><br>
+     * <ul>
+     * <li><code>null</code> - The engine implementation is not thread safe, and cannot
+     * be used to execute scripts concurrently on multiple threads.
+     * <li><code>&quot;MULTITHREADED&quot;</code> - The engine implementation is internally
+     * thread-safe and scripts may execute concurrently although effects of script execution
+     * on one thread may be visible to scripts on other threads.
+     * <li><code>&quot;THREAD-ISOLATED&quot;</code> - The implementation satisfies the requirements
+     * of &quot;MULTITHREADED&quot;, and also, the engine maintains independent values
+     * for symbols in scripts executing on different threads.
+     * <li><code>&quot;STATELESS&quot;</code> - The implementation satisfies the requirements of
+     * <li><code>&quot;THREAD-ISOLATED&quot;</code>.  In addition, script executions do not alter the
+     * mappings in the <code>Bindings</code> which is the engine scope of the
+     * <code>ScriptEngine</code>.  In particular, the keys in the <code>Bindings</code>
+     * and their associated values are the same before and after the execution of the script.
+     * </ul>
+     * <br><br>
+     * Implementations may define implementation-specific keys.
+     *
+     * @param key The name of the parameter
+     * @return The value for the given parameter. Returns <code>null</code> if no
+     * value is assigned to the key.
+     *
+     */
+    public @Nullable Object getParameter(@Nullable String key);
```

Comment:
The argument cannot be null.  (The returned value might be, but the documentation does not permit the argument to be null.)